src/main/kotlin/no/nav/tpts/mottak/joark/JoarkConsumer.kt

```diff
@@ -44,3 +57,89 @@ fun createJoarkConsumer(topicName: String): KafkaConsumer<String, GenericRecord>
         }
     ).also { it.subscribe(listOf(topicName)) }
 }
+
+internal class JoarkConsumer(private val consumer: Consumer<String, GenericRecord>) : CoroutineScope {
+    override val coroutineContext: CoroutineContext
+        get() = Dispatchers.IO + job
+    private val job: Job = Job()
+
+    init {
+        Runtime.getRuntime().addShutdownHook(Thread(::shutdownHook))
+    }
+
+    fun start() {
+        LOG.info { "starting JoarkConsumer" }
+        launch {
```

Comment:
Hvis jeg forstår denne koden riktig, så jeg man kunne evt gjort noe ala dette:
`val job = launch(Dispatchers.IO) { ... }`
Istedetfor å extende CoroutineScope så override corourineContext, synes ikke det er helt lett å forstå hva det innebærer å "plusse" en CoroutineContext med en Job

Reply:
Om klassen ikke extender CouroutineContext, da må vel alle metodene være suspend da? 
I og for seg enig med deg når det gjelder `Dispatchers.IO + job`. Ikke så lett å forstå hvorfor det er sånn. Å launche i IO kontekst + job gjorde at selv konsumeren også stopper ved cancel. Men burde kanskje skrevet test på det?